pages/api/authenticate.ts

```diff
@@ -0,0 +1,42 @@
+import type {NextApiRequest, NextApiResponse} from 'next';
+
+import {PrivyClient} from '@privy-io/server-auth';
+
+const PRIVY_APP_ID = process.env.NEXT_PUBLIC_PRIVY_APP_ID;
+const PRIVY_APP_SECRET = process.env.PRIVY_APP_SECRET;
+const client = new PrivyClient(PRIVY_APP_ID as string, PRIVY_APP_SECRET as string);
+
+export type AuthenticateSuccessResponse = {
+  claims: {
+    appId: string,
+    issuer: string,
+    issuedAt: number,
+    expiration: number,
+    sessionId: string,
+    userId: string
+  }
+};
+
+export type AuthenticationErrorResponse = {
+  error: string;
+};
+
+export default async function GET(req: NextApiRequest, res: NextApiResponse<AuthenticateSuccessResponse | AuthenticationErrorResponse>) {
```

Comment:
nit, I would rename this `handler` the `GET`/`POST` etc. naming is an app router paradigm and this endpoint will get called for any http method at that path. Nothing wrong about this example just not very idiomatic